### PR TITLE
issue/6473-document-api-arguments

### LINKED ISSUE #6473: Document all API arguments
This ticket should make sure that all the arguments of all API endpoints are documented in the `methods.py` and `methods_v2.py`.

Initial timebox: 1 day

### PR DESCRIPTION
**Pull request by FloLey**
# Description

document all the missing api arguments

closes #6473 
# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [x] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

File: src/inmanta/protocol/methods.py [hunk 1]
```diff
@@ -114,22 +121,22 @@ def create_environment(
     Create a new environment
 
     :param project_id: The id of the project this environment belongs to
-    :param name: The name of the environment
-    :param repository: The url (in git form) of the repository
-    :param branch: The name of the branch in the repository
-    :param environment_id: A unique environment id, if none an id is allocated by the server
+    :param name: The name of the environment.
+    :param repository: Optional. The URL (in git format) of the repository.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:125]: What does `git format` mean?

File: src/inmanta/protocol/methods.py [hunk 2]
```diff
@@ -114,22 +121,22 @@ def create_environment(
...
+    Modify the given environment.
 
-    :param id: The id of the environment
-    :param name: The name of the environment
-    :param repository: The url (in git form) of the repository
-    :param branch: The name of the branch in the repository
+    :param id: The id of the environment to modify.
+    :param name: The new name for the environment.
+    :param repository: Optional. The URL (in git format) of the repository.
+    :param branch: Optional. The name of the branch in the repository.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:139]: We should mention here what the behavior is when branch or repository is set to None.

File: src/inmanta/protocol/methods.py [hunk 3]
```diff
@@ -160,11 +167,11 @@ def list_environments():
...
     """
-    Get an environment and all versions associated
+    Get an environment and all versions associated.
 
-    :param id: The id of the environment to return
-    :param versions: Include this many available version for this environment.
-    :param resources: Include this many available resources for this environment.
+    :param id: The id of the environment to return.
+    :param versions: Optional. Include this many available versions for this environment.
+    :param resources: Optional. Include this many available resources for this environment.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:174]: When I look at the implementation, this is not really true, is it?

File: src/inmanta/protocol/methods.py [hunk 4]
```diff
@@ -224,7 +240,10 @@ def get_setting(tid: uuid.UUID, id: str):
 )
 def delete_setting(tid: uuid.UUID, id: str):
     """
-    Delete a value
+    Delete a setting.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:243]: You need to be more precise here about what deleting a setting means.

File: src/inmanta/protocol/methods.py [hunk 5]
```diff
@@ -259,7 +277,7 @@ def clear_environment(id: uuid.UUID):
     """
     Clear all data from this environment.
 
-    :param id: The uuid of the environment.
+    :param id: The id of the environment to be cleared.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:280]: We need to be more precise in the description of this endpoint. Not all data is removed by this endpoint. The settings are kept.

File: src/inmanta/protocol/methods.py [hunk 6]
```diff
@@ -785,6 +806,7 @@ def get_code(tid: uuid.UUID, id: int, resource: str):
 
     :param tid: The environment the code belongs to
     :param id: The id (version) of the configuration model
+    :param resource: The id of the resource to get the code from
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:809]: We should also make clear whether this is a resource id or a resource version id.
```suggestion
    :param resource: The id of the resource to get the code for
```

File: src/inmanta/protocol/methods.py [hunk 7]
```diff
@@ -795,17 +817,22 @@ def upload_code_batched(tid: uuid.UUID, id: int, resources: dict):
 
     :param tid: The environment the code belongs to
     :param id: The id (version) of the configuration model
-    :param resource: a dict mapping resources to dicts mapping file names to file hashes
+    :param resources: a dict mapping resources to dicts mapping file names to file hashes
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:820]: We should be more specific about the key here. What does "mapping reosurces" mean?

File: src/inmanta/protocol/methods.py [hunk 8]
```diff
@@ -818,9 +845,9 @@ def get_reports(tid: uuid.UUID, start: str = None, end: str = None, limit: int =
     Return compile reports newer then start
 
     :param tid: The id of the environment to get a report from
-    :param start: Reports after start
-    :param end: Reports before end
-    :param limit: Maximum number of results, up to a maximum of 1000
+    :param start: Optional. Reports after start
+    :param end: Optional. Reports before end
+    :param limit: Optional. Maximum number of results, up to a maximum of 1000
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:850]: We should mention the default behavior when the limit is None. The same of the other endpoints with this limit parameter.

File: src/inmanta/protocol/methods.py [hunk 9]
```diff
@@ -912,6 +939,9 @@ def get_status():
 def set_state(agent: str, enabled: bool):
     """
     Set the state of the agent.
+
+    :param agent: The name of the agent.
+    :param enabled: A boolean value indicating whether the agent should be enabled or disabled.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:944]:
```suggestion
    :param enabled: A boolean value indicating whether the agent should be paused (enabled=False) or unpaused (enabled=True).
```

File: src/inmanta/server/services/fileservice.py [hunk 10]
```diff
@@ -94,7 +94,7 @@ async def stat_file_internal(self, files: Iterable[str]) -> List[str]:
         return list(await File.get_non_existing_files(files))
 
     @handle(methods.diff)
-    async def file_diff(self, a: str, b: str) -> Apireturn:
+    async def file_diff(self, file_id_1: str, file_id_2: str) -> Apireturn:
```
Comment (arnaudsjs) [inline src/inmanta/server/services/fileservice.py:97]: I like this change a lot

File: src/inmanta/server/services/fileservice.py [hunk 11]
```diff
@@ -111,10 +111,10 @@ async def _get_lines_for_file(content_hash: str) -> list[str]:
                 # keepends for backwards compatibility with <file_handle>.readlines()
                 return file_content.splitlines(keepends=True)
 
-        a_lines = await _get_lines_for_file(content_hash=a)
-        b_lines = await _get_lines_for_file(content_hash=b)
+        a_lines = await _get_lines_for_file(content_hash=file_id_1)
+        b_lines = await _get_lines_for_file(content_hash=file_id_2)
```
Comment (arnaudsjs) [inline src/inmanta/server/services/fileservice.py:115]:
```suggestion
        file_1_lines = await _get_lines_for_file(content_hash=file_id_1)
        file_2_lines = await _get_lines_for_file(content_hash=file_id_2)
```

File: src/inmanta/server/services/fileservice.py [hunk 12]
```diff
@@ -111,10 +111,10 @@ async def _get_lines_for_file(content_hash: str) -> list[str]:
...
         try:
-            diff = difflib.unified_diff(a_lines, b_lines, fromfile=a, tofile=b)
+            diff = difflib.unified_diff(a_lines, b_lines, fromfile=file_id_1, tofile=file_id_2)
```
Comment (arnaudsjs) [inline src/inmanta/server/services/fileservice.py:117]:
```suggestion
            diff = difflib.unified_diff(file_1_lines, file_2_lines, fromfile=file_id_1, tofile=file_id_2)
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/protocol/methods.py (hunk 1 above)
Comment (FloLey) [reply src/inmanta/protocol/methods.py:125]: I mean ending in .git. I'm not sure how to express this better

Comment (arnaudsjs) [reply src/inmanta/protocol/methods.py:125]: The fact that URLs of Git repositories end with `.git` is more a convention than hard requirement. Our API also doesn't enforce it. So I wouldn't mention is in our documentation either.

File: src/inmanta/protocol/methods_v2.py [hunk 13]
```diff
@@ -88,20 +88,26 @@ def project_create(name: str, project_id: uuid.UUID = None) -> model.Project:
 def project_modify(id: uuid.UUID, name: str) -> model.Project:
     """
     Modify the given project
```
Comment (Hugo-Inmanta) [inline src/inmanta/protocol/methods_v2.py:90]:
```suggestion
    Rename the given project
```

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

File: src/inmanta/protocol/methods.py [hunk 14]
```diff
@@ -228,7 +250,11 @@ def get_setting(tid: uuid.UUID, id: str):
 )
 def delete_setting(tid: uuid.UUID, id: str):
     """
-    Delete a value
+    Delete a setting identified by the 'key' from the specified environment.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:253]:
```suggestion
    Restore the given setting to its default value.
```

File: src/inmanta/protocol/methods.py [hunk 15]
```diff
@@ -794,31 +828,42 @@ def get_parameter(tid: uuid.UUID, agent: str, resource: dict):
...
+    :param id: The version number of the configuration model.
+    :param resources: A dictionary where each key is a string representing a resource type.
+                  For each resource type, the value is a dictionary. This nested dictionary's keys are file names,
+                  and each key maps to a tuple. This tuple contains three elements: the file name, the module name,
+                  and a list of requirements.
 
-    :param tid: The environment the code belongs to
-    :param id: The id (version) of the configuration model
-    :param resource: a dict mapping resources to dicts mapping file names to file hashes
+    The method validates that all provided file references are valid and checks for conflicts with existing code entries.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods.py:851]:
```suggestion
    The endpoint validates that all provided file references are valid and checks for conflicts with existing code entries.
```

File: src/inmanta/protocol/methods_v2.py [hunk 16]
```diff
@@ -298,6 +307,8 @@ def environment_create_token(tid: uuid.UUID, client_types: List[str], idempotent
 )
 def environment_settings_list(tid: uuid.UUID) -> model.EnvironmentSettingsReponse:
     """
+    :param tid: The id of the environment for which the list of settings is being requested.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods_v2.py:310]: Does this render properly in the API documentation if we put the parameters first and only the the description?

File: src/inmanta/protocol/methods_v2.py [hunk 17]
```diff
@@ -343,7 +363,10 @@ def environment_setting_get(tid: uuid.UUID, id: str) -> model.EnvironmentSetting
 )
 def environment_setting_delete(tid: uuid.UUID, id: str) -> ReturnValue[None]:
     """
-    Delete a value
+    Delete a specific setting from an environment's configuration.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods_v2.py:366]:
```suggestion
    Reset the given setting to its default value.
```

File: src/inmanta/protocol/methods_v2.py [hunk 18]
```diff
@@ -313,7 +324,12 @@ def environment_settings_list(tid: uuid.UUID) -> model.EnvironmentSettingsRepons
 )
 def environment_settings_set(tid: uuid.UUID, id: str, value: model.EnvSettingType) -> ReturnValue[None]:
     """
-    Set a value
+    Set or update a specific setting in an environment's configuration.
```
Comment (arnaudsjs) [inline src/inmanta/protocol/methods_v2.py:327]: I don't really see what the difference would be between set and update.
```suggestion
    Set a specific setting in an environment's configuration.
```

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: src/inmanta/protocol/methods_v2.py (hunk 16 above)
Comment (FloLey) [reply src/inmanta/protocol/methods_v2.py:310]: I didn't check but I don't like it, it is not consistent with the rest. I will change it

File: src/inmanta/protocol/methods.py [hunk 19]
```diff
@@ -160,11 +169,15 @@ def list_environments():
...
-    :param id: The id of the environment to return
-    :param versions: Include this many available version for this environment.
-    :param resources: Include this many available resources for this environment.
+    :param id: The id of the environment to return.
+    :param versions: Optional. If provided and greater than 0, include this many of the most recent versions for this
+                     environment, ordered in descending order of their version number.
+                     If not provided or 0, no version information is included.
+    :param resources: Optional. If provided and greater than 0, include a report of all resources in the environment,
+                     The number of resources included in the report is up to the specified 'resources' value.
+                     If not provided or 0, no resource information is included.
```
[the remark this answers is not in the document]
Comment (FloLey) [reply src/inmanta/protocol/methods.py:184]: indeed, I thought resources and versions behaved the same way. I'll change it

[the remark this answers is not in the document]
Comment (FloLey) [reply src/inmanta/protocol/methods.py]: Do you now why it isn't a bool?

[the remark this answers is not in the document]
Comment (arnaudsjs) [reply src/inmanta/protocol/methods.py]: No idea.

**Review by arnaudsjs — APPROVED** [review]

**Review by Hugo-Inmanta — APPROVED** [review]

**Pull request closed without merging**